pandas/tests/test_resample.py

```diff
@@ -2939,6 +2939,14 @@ def test_resample_with_nat(self):
                                                    freq='1S'))
         assert_frame_equal(result, expected)
 
+    def test_resample_as_freq_with_subperiod(self):
+        # GH 13022
+        index = pd.timedelta_range('00:00:00', '00:10:00', freq='5T')
+        df = DataFrame(data=[[1, 3], [-5, 10], [0, 0]], index=index)
+        result = df.resample('1T').asfreq()
```

Comment:
Instead, could you construct the expected frame and use `tm.assert_frame_equal`?